### PR TITLE
fix/payment-gas-estimation

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- New integration tests
- Update rif-relay-client
- Update rif-relay-server

## Why

- RSKj Arrowhead version have some changes in the gas cost
- New smart wallet type

## Refs

- https://github.com/rsksmart/rif-relay-client/pull/91
- https://github.com/rsksmart/rif-relay-server/pull/148

### CODE REVIEW

**Review by antomor — APPROVED** [review]
Due to the nature of these changes, they're fine, but as we discussed, if we had to work on it in future, we may need to refactor these tests a bit.

File: test/utils/NoClient.ts [hunk 1]
```diff
@@ -470,7 +470,7 @@ const isContractCallInvalid = (
 ): boolean => {
   return (
     to != constants.AddressZero &&
-    data === '0x00' &&
+    isDataEmpty(data.toString()) &&
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply test/utils/NoClient.ts:473]: Take into account that the client is in charge of trying to generate the request, so it needs to be sure that the information sent by the user, has a valid format.

File: hardhat.config.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ const config: HardhatUserConfig = {
     regtest: {
       url: getRskNodeUrl(),
       chainId: 33,
+      gasPrice: 1, //RSKj on startup returns 0 gasPrice on first tx.
```
Comment (antomor) [inline hardhat.config.ts:38]: 👏🏽👏🏽👏🏽

**Review by antomor — APPROVED** [review]
@franciscotobar LGTM, thank you 🙏🏼 !

**Pull request merged**